src/contracts/core/RewardsCoordinator.sol

```diff
@@ -340,6 +349,83 @@ contract RewardsCoordinator is
+        );
+
+        uint256 totalAmount = 0;
+        address currOperatorAddress = address(0);
+        for (uint256 i = 0; i < performanceRewardsSubmission.operatorRewards.length; ++i) {
+            OperatorReward calldata operatorReward = performanceRewardsSubmission.operatorRewards[i];
+            require(
+                operatorReward.operator != address(0),
+                "RewardsCoordinator._validatePerformanceRewardsSubmission: operator cannot be 0 address"
+            );
+            require(
+                currOperatorAddress < operatorReward.operator,
+                "RewardsCoordinator._validatePerformanceRewardsSubmission: operators must be in ascending order to handle duplicates"
+            );
+            currOperatorAddress = operatorReward.operator;
+            require(
+                operatorReward.amount > 0,
+                "RewardsCoordinator._validatePerformanceRewardsSubmission: operator reward amount cannot be 0"
+            );
+            totalAmount += operatorReward.amount;
+        }
+
+        require(
+            performanceRewardsSubmission.duration <= MAX_REWARDS_DURATION,
+            "RewardsCoordinator._validatePerformanceRewardsSubmission: duration exceeds MAX_REWARDS_DURATION"
+        );
+        require(
+            performanceRewardsSubmission.duration % CALCULATION_INTERVAL_SECONDS == 0,
+            "RewardsCoordinator._validatePerformanceRewardsSubmission: duration must be a multiple of CALCULATION_INTERVAL_SECONDS"
+        );
+        require(
+            performanceRewardsSubmission.startTimestamp % CALCULATION_INTERVAL_SECONDS == 0,
+            "RewardsCoordinator._validatePerformanceRewardsSubmission: startTimestamp must be a multiple of CALCULATION_INTERVAL_SECONDS"
+        );
+        require(
+            block.timestamp - MAX_RETROACTIVE_LENGTH <= performanceRewardsSubmission.startTimestamp &&
+                GENESIS_REWARDS_TIMESTAMP <= performanceRewardsSubmission.startTimestamp,
+            "RewardsCoordinator._validatePerformanceRewardsSubmission: startTimestamp too far in the past"
+        );
+        require(
+            performanceRewardsSubmission.startTimestamp + performanceRewardsSubmission.duration < block.timestamp,
+            "RewardsCoordinator._validatePerformanceRewardsSubmission: performance rewards submission is not retroactive"
+        );
+
+        // Require performanceRewardsSubmission is for whitelisted strategy or beaconChainETHStrategy
+        address currAddress = address(0);
+        for (uint256 i = 0; i < performanceRewardsSubmission.strategiesAndMultipliers.length; ++i) {
+            IStrategy strategy = performanceRewardsSubmission.strategiesAndMultipliers[i].strategy;
+            require(
+                strategyManager.strategyIsWhitelistedForDeposit(strategy) || strategy == beaconChainETHStrategy,
+                "RewardsCoordinator._validatePerformanceRewardsSubmission: invalid strategy considered"
+            );
```

Comment:
does this propertly support longtail assets?

Reply:
Good question. It was something I had noted down myself to check.

Reply:
Rewards works out of the box for longtail assets without overflow since we limit the number of shares that can be minted for any strategy

Reply:
@ypatil12 Are you saying that we don't need this check?

Reply:
Got it! Resolving this thread.